pkg/proxy/iptables/proxier.go

```diff
@@ -518,6 +518,7 @@ func (proxier *Proxier) SyncLoop() {
 	defer t.Stop()
 	// Update healthz timestamp at beginning in case Sync() never succeeds.
 	if proxier.healthzServer != nil {
+		registerMetrics()
```

Comment:
Because the metrics [handleFuncs](https://github.com/kubernetes/kubernetes/pull/45889/commits/16c41e3266b56d6ca67915c00d58af6293459cd2#diff-cf93bbed37202b7d58b5841f07ddd89fR565) are added to metrics server but not healthz server, I think we shouldn't put it in this condition block.